website/content/middleware/timeout.md

```diff
@@ -58,6 +72,64 @@ TimeoutErrorHandlerWithContext func(error, echo.Context) error
 DefaultTimeoutConfig = TimeoutConfig{
     Skipper:      DefaultSkipper,
     Timeout:      0,
-    ErrorHandler: nil,
+    ErrorMessage: "",
+}
+```
+
+## Alternatively handle timeouts in handlers
+
+```go
+func main() {
+    e := echo.New()
+
+    doBusinessLogic := func(ctx context.Context, UID string) error {
+        // NB: Do not use echo.JSON() or any other method that writes data/headers to client here. This function is executed
+        // in different coroutine that should not access echo.Context and response writer
+
+        log.Printf("uid: %v\n", UID)
+        //res, err := slowDatabaseCon.ExecContext(ctx, query, args)
+        time.Sleep(10 * time.Second) // simulate slow execution
+        log.Print("doBusinessLogic done\n")
+        return nil
+    }
+
+    handlerFunc := func(c echo.Context) error {
+        defer log.Print("handlerFunc done\n")
+
+        // extract and validate needed data from request and pass it to business function
+        UID := c.QueryParam("uid")
+
+        ctx, cancel := context.WithTimeout(c.Request().Context(), 5 * time.Second)
+        defer cancel()
+        result := make(chan error)
+        go func() { // run actual business logic in separate coroutine
+            defer func() { // unhandled panic in coroutine will crash the whole application
+                if err := recover(); err != nil {
+                    result <- fmt.Errorf("panic: %v", err)
+                }
+            }()
+            result <- doBusinessLogic(ctx, UID)
+        }()
+
+        select { // wait until doBusinessLogic finishes or we timeout while waiting for the result
+        case <-ctx.Done():
+            err := ctx.Err()
+            if err == context.DeadlineExceeded {
+                return echo.NewHTTPError(http.StatusServiceUnavailable, "doBusinessLogic timeout")
+            }
+            return err // probably client closed the connection
+        case err := <-result: // doBusinessLogic finishes
+            if err != nil {
+                return err
+            }
+        }
+        return c.NoContent(http.StatusAccepted)
```

Comment:
If I got it correctly, this is returning a 202 even when doBusinessLogic finished. At this point the code should return 200 instead.